src/main/java/raythx98/duke/Duke.java

```diff
@@ -0,0 +1,45 @@
+package raythx98.duke;
+
+import raythx98.duke.command.Command;
+import raythx98.duke.exception.DukeException;
+import raythx98.duke.parser.Parser;
+import raythx98.duke.storage.Storage;
+import raythx98.duke.task.TaskList;
+import raythx98.duke.ui.Ui;
+
+/**
+ * Main driving force for the Duke application.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    public Duke(String filePath) {
+        ui = new Ui();
```

Comment:
consider referencing instance level variables with `this.` to mitigate variable shadowing and be clearer

Reply:
Look at:
https://se-education.org/guides/conventions/java/index.html
under variables where it states "Use the this keyword only when a field is shadowed by a method or constructor parameter."

Also, please leave only 1 review (with multiple comments) to avoid spamming the contributor's inbox, thank you :)